apps/citizen-func/utils/redis_storage.ts

```diff
@@ -2,204 +2,70 @@ import * as E from "fp-ts/lib/Either";
 import { pipe } from "fp-ts/lib/function";
 import * as O from "fp-ts/lib/Option";
 import * as TE from "fp-ts/lib/TaskEither";
-import { RedisClient } from "redis";
-
-/**
- * Parse a Redis given string reply.
- *
- * @see https://redis.io/topics/protocol#simple-string-reply.
- */
-const givenStringReply = (
-  err: Error | null,
-  reply: string | undefined,
-  message: string
-): E.Either<Error, boolean> =>
-  pipe(
-    err,
-    O.fromNullable,
-    O.map(E.left),
-    O.getOrElse(() => E.right(reply === message))
-  );
-
-/**
- * Parse a Redis single string reply.
- *
- * @see https://redis.io/topics/protocol#simple-string-reply.
- */
-const singleStringReply = (
-  err: Error | null,
-  reply: "OK" | undefined
-): E.Either<Error, boolean> =>
-  pipe(
-    err,
-    O.fromNullable,
-    O.map(E.left),
-    O.getOrElse(() => E.right(reply === "OK"))
-  );
-
-/**
- * Parse a Redis single string reply.
- *
- * @see https://redis.io/topics/protocol#simple-string-reply.
- */
-const singleValueReply = (
-  err: Error | null,
-  reply: string | null
-): E.Either<Error, O.Option<string>> =>
-  pipe(
-    err,
-    O.fromNullable,
-    O.map(E.left),
-    O.getOrElse(() => E.right(O.fromNullable(reply)))
-  );
-
-/**
- * Parse a Redis integer reply.
- *
- * @see https://redis.io/topics/protocol#integer-reply
- */
-const integerReply = (
-  err: Error | null,
-  reply: unknown,
-  expectedReply?: number
-): E.Either<Error, boolean> => {
-  if (err) {
-    return E.left<Error, boolean>(err);
-  }
-  if (expectedReply !== undefined && expectedReply !== reply) {
-    return E.right<Error, boolean>(false);
-  }
-  return E.right<Error, boolean>(typeof reply === "number");
-};
-
-const isTrue = (b: boolean): b is true => b === true;
-
-const falsyResponseToError = (
-  response: E.Either<Error, boolean>,
-  error: Error
-): E.Either<Error, true> =>
-  pipe(
-    response,
-    E.filterOrElse(isTrue, () => error)
-  );
+import * as redis from "redis";
 
 export const setWithExpirationTask = (
-  redisClient: RedisClient,
+  redisClient: redis.RedisClientType,
   key: string,
   value: string,
   expirationInSeconds: number,
   errorMsg?: string
 ): TE.TaskEither<Error, true> =>
   pipe(
     TE.tryCatch(
+      () => redisClient.set(key, value, { EX: expirationInSeconds }),
```

Comment:
I think the new redis client have something like `setEx` command. 
Just see here https://github.com/search?q=repo%3Apagopa%2Fio-backend+setEx&type=code

Reply:
@michaeldisaro If I am not mistaken the `setEx` method you are mentioning here is for redis-cluster, i passed EX as part of `SetOption` as is indicated [here](https://github.com/redis/node-redis/blob/b884e132fa54908b022f565263d1402b43cec83a/packages/client/lib/commands/SET.ts#L27)

Reply:
You could also be right from docs, but we used `setEx` in many upgrades of redis till today https://github.com/search?q=org%3Apagopa+setEx%28&type=code 
I'd prefer to maintain coherence with other repos because some are really recent and not old upgraded once (see fast-login).

Reply:
Ok @michaeldisaro understood, I did the fix you requested here [e5c15f3](https://github.com/pagopa/io-messages/pull/12/commits/e5c15f339a440d276dc874f2231f655e1fd79a53), also please check those changes to unit tests to ensure that the behaviour si correct [1ca5dbe](https://github.com/pagopa/io-messages/pull/12/commits/1ca5dbe30372690797adea7a968e691d79329998)

Reply:
LGTM, new redis commands returns promises, so tests look legit to me.

Reply:
@michaeldisaro I also removed `jest-mock-express`  here [212219d](https://github.com/pagopa/io-messages/pull/12/commits/212219d27d3725011f89c0750ef5034496692466) cause it was unused as well, we got this unused dep also in `sending-func`